crates/engine/tree/benches/state_root_task.rs

```diff
@@ -225,53 +217,21 @@ fn bench_state_root(c: &mut Criterion) {
                                 .expect("Failed to create proof worker thread pool"),
                         );
 
-                        (
-                            config,
-                            state_updates,
-                            provider,
-                            nodes_sorted,
-                            state_sorted,
-                            prefix_sets,
-                            state_root_task_pool,
-                        )
+                        (config, state_updates, state_root_task_pool)
                     },
-                    |(
-                        config,
-                        state_updates,
-                        provider,
-                        nodes_sorted,
-                        state_sorted,
-                        prefix_sets,
-                        state_root_task_pool,
-                    )| {
-                        let blinded_provider_factory = ProofBlindedProviderFactory::new(
-                            InMemoryTrieCursorFactory::new(
-                                DatabaseTrieCursorFactory::new(provider.tx_ref()),
-                                &nodes_sorted,
-                            ),
-                            HashedPostStateCursorFactory::new(
-                                DatabaseHashedCursorFactory::new(provider.tx_ref()),
-                                &state_sorted,
-                            ),
-                            prefix_sets,
-                        );
-
-                        black_box(std::thread::scope(|scope| {
-                            let task = StateRootTask::new(
-                                config,
-                                blinded_provider_factory,
-                                state_root_task_pool,
-                            );
+                    |(config, state_updates, state_root_task_pool)| {
+                        black_box({
```

Comment:
return value of the closure is already black_boxed, so just return something from it instead of wrapping everything in one